canvas/cli/common.py

```diff
@@ -1,35 +1,49 @@
+import json
 import sys
 
+DEFAULT_JSON = False
+DEFAULT_TABLE = False
+
 # keys: [(items key, title, pretty title), ...]
-def cli_list(items, keys, table = False, skip_headers = False,
-        collective_name = 'items', sort_key = 'id'):
+def cli_list(items, keys, table = DEFAULT_TABLE, skip_headers = False,
+        collective_name = 'items', sort_key = 'id', output_json = DEFAULT_JSON):
     if (len(items) == 0):
         print("No %s found." % (collective_name), file = sys.stderr)
         return 0
 
-    if (table and (not skip_headers)):
-        print("\t".join([key_set[1] for key_set in keys]))
-
-    items = list(sorted(items, key = lambda item: item.get(sort_key, '')))
-    for i in range(len(items)):
-        item = items[i]
+    items = list(sorted(items, key=lambda item: item.get(sort_key, '')))
 
+    processed_items = []
+    for item in items:
         values = {}
+
         for items_key, _, _ in keys:
             value = item[items_key]
             if (value is None):
                 value = ''
 
-            values[items_key] = str(value).strip()
+            if (output_json):
+                values[items_key] = value
+            else:
+                values[items_key] = str(value).strip()
+
+        processed_items.append(values)
 
-        if (table):
-            print("\t".join(map(_clean_cell, [values[key_set[0]] for key_set in keys])))
-        else:
-            if (i != 0):
-                print()
+    if (output_json):
+        print(json.dumps(processed_items, indent = 4, sort_keys = True))
+    else:
```

Comment:
I generally don't like large conditionals,
they make code more complex and harder to read.

Instead, return when you can.
For example, the JSON branch doesn't need anything after it prints.

It may also be cleaner to handle each case in a conditional instead of grouping together table and non-table.